syncer/sharding_group.go

```diff
@@ -313,23 +317,32 @@ func (sg *ShardingGroup) FirstLocationUnresolved() *binlog.Location {
 	defer sg.RUnlock()
 	if sg.remain < len(sg.sources) && sg.firstLocation != nil {
 		// create a new pos to return
+		gset := gtid.MinGTIDSet(sg.flavor)
+		if sg.firstLocation.GTIDSet != nil {
+			gset = sg.firstLocation.GTIDSet.Clone()
+		}
+
 		return &binlog.Location{
 			Position: mysql.Position{
 				Name: sg.firstLocation.Position.Name,
 				Pos:  sg.firstLocation.Position.Pos,
 			},
-			GTIDSet: sg.firstLocation.GTIDSet.Clone(),
+			GTIDSet: gset,
 		}
 	}
 	item := sg.meta.GetGlobalActiveDDL()
 	if item != nil {
 		// make a new copy
+		gset := gtid.MinGTIDSet(sg.flavor)
+		if item.FirstLocation.GTIDSet != nil {
+			gset = item.FirstLocation.GTIDSet.Clone()
+		}
```

Comment:
may a helper function can be added to do this job for reuse (_check is nil_, _create min GSet_ and _do the copy_).

Reply:
add in c5bfffaa928f9a81094faa6a37216a79279d65f0